### ISSUE TITLE
Ignore whitespace in bindings

### ISSUE THREAD

**Issue by Y-Less** [angular/angular#3108]
I have this component template:

```html
<div class="alert" role="alert"
  [class.alert-success]="getAlertType() == 3"
  [class.alert-info]="getAlertType() == 2"
  [class.alert-warning]="getAlertType() == 1"
  [class.alert-danger]="getAlertType() == 0"
  [class.alert-dismissable]="isDismissable()"
>
    <button *ng-if="isDismissable()" type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button>
    <content></content>
</div>
```

For aesthetics, I originally wrote it like this:

```html
<div class="alert" role="alert"
  [class.alert-success]="getAlertType() == 3"
  [class.alert-info   ]="getAlertType() == 2"
  [class.alert-warning]="getAlertType() == 1"
  [class.alert-danger ]="getAlertType() == 0"
  [class.alert-dismissable]="isDismissable()"
>
    <button *ng-if="isDismissable()" type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button>
    <content></content>
</div>
```

Obviously so that all the similar tests line up, but done like this they are not set correctly until I delete the trailing spaces.  I also don't know if there is a better way to do this, but I'm still playing.

**Comment by gdi2290** [comment]
yeah that does make sense

**Comment by pkozlowski-opensource** [comment]
This kind of does make sense but I'm not sure we can do much about it since if you type `<div [class.alert-info   ]="getAlertType() == 2"></div>` a browser sees it as a `<div>` with _2_ attributes:
* `[class.alert-info`, no value
* `]` with the `getAlertType() == 2` value

So this is kind of "broken" by a browser before AngularJS can do anything about it... The only way of "solving" this issue would be to parse template strings instead of live DOM already parsed by a browser, but even in this case we are kind of breaking attributes naming convention.

In short: you can't have spaces in attribute names (for example, you can't write `foo-  bar="baz"` instead of ``foo-bar="baz"`. Also you can't write `bind-class.alert-info  ="getAlertType() == 2"`

Going to close this one for now as I don't think we can / should do much here.
